File: wesafe.net/WeSafe.DAL.Entities/Device.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using WeSafe.Shared.Enumerations;

namespace WeSafe.DAL.Entities
{
    public class Device
    {
        public int Id { get; set; }

        public DeviceType DeviceType { get; set; }

        public string SerialNumber { get; set; }

        public string HWVersion { get; set; }

        public string SWVersion { get; set; }

        public string NVIDIASn { get; set; }

        public string MACAddress { get; set; }

        public int? ClientId { get; set; }

        public virtual Client Client { get; set; }

        public string ClientNetworkIp { get; set; }

        public DateTimeOffset? ActivationDate { get; set; }

        public DateTimeOffset? AssemblingDate { get; set; }

        public string CreatedBy { get; set; }

        [ForeignKey("CreatedBy")]
        public User User { get; set; }

        public string Info { get; set; }

        public string Token { get; set; }

        public string Status { get; set; }

        public string NetworkStatus { get; set; }

        public ICollection<Camera> Cameras { get; set; }

        public bool IsArmed { get; set; }

        public string Name { get; set; }

        public string CurrentSshPassword { get; set; }

        public string PreviousSshPassword { get; set; }

        public DateTime LastUpdateDatePassword { get; set; }

        public ICollection<DeviceLog> DeviceLogs { get; set; }

        public string AuthToken { get; set; }

        public int? MaxActiveCameras { get; set; }

        public string TimeZone { get; set; }

        public ICollection<ClientSubscriberAssignment> Assignments { get; set; }

        public ICollection<DeviceIndicators> Indicators { get; set; }

        public ICollection<DetectedCamera> DetectedCameras { get; set; }

        public string Metadata { get; set; }
    }
}

File: wesafe.net/WeSafe.WebApi.Light/ClientApp/src/app/models/device.ts
import { ICamera } from './camera';

export interface IDevice {
    id: number;
    serialNumber: string;
    hwVersion: string;
    swVersion: string;
    nvidiaSn: string;
    macAddress: string;
    clientId: number;
    clientName: string;
    clientNetworkIp: string;
    activationDate: string;
    assemblingDate: string;
    createdBy: string;
    info: string;
    token: string;
    status: string;
    isArmed: boolean;
    camerasNumber: number;
    name: string;
    maxActiveCameras: number;
    cameras?: Array<ICamera>;
}

export class Device implements IDevice {
    id: number;
    activationDate: string;
    assemblingDate: string;
    clientId: number;
    clientName: string;
    clientNetworkIp: string;
    createdBy: string;
    hwVersion: string;
    info: string;
    macAddress: string;
    nvidiaSn: string;
    swVersion: string;
    serialNumber: string;
    token: string;
    status: string;
    isArmed: boolean;
    camerasNumber: number;
    name: string;
    maxActiveCameras: number;

    constructor(obj?: IDevice) {
        if (obj) Object.assign(this, obj);
    }
}

export interface IDeviceType {
    id: number;
    value: string;
}

export class DeviceType implements IDeviceType {
    id: number;
    value: string;
}
